### PR TITLE
docs(docs-infra): replace `jszip` with `fflate`

### LINKED ISSUE #56404: Hello World Project is not Downloading from website.
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

Yes

### Description

I am Learning angular and when i open angular docs and then i try to download Hello World project. But it was not download. then i check the console of my browser and found error.
setImmediate is not defined

### Please provide a link to a minimal reproduction of the bug

Image link: https://drive.google.com/file/d/1Q8_cpPOkGE_A1ZI7GFhCzS3ckTt2FMxF/view?usp=sharing

### Please provide the exception or error you saw

_No response_

### Please provide the environment you discovered this bug in (run `ng version`)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by JeanMeche**
Fixes #56404

Note: The zip still contains some file that shouldn't be there but this will be covered with #56739

### CODE REVIEW

File: adev/BUILD.bazel [hunk 1]
```diff
@@ -157,6 +157,7 @@ architect(
         "--poll=1000",
         "--live-reload",
         "--watch",
+        "--verbose",
```
Comment (JeanMeche) [inline adev/BUILD.bazel:160]: This is the only way I found to log the build issues. 
Without this, when the build fails, it fails silently.

File: adev/src/app/editor/download-manager.service.ts [hunk 2]
```diff
@@ -50,4 +51,20 @@ export class DownloadManager {
...
+    });
+
+    console.log(filesObj);
+    const content = new Promise<Uint8Array>((resolve) => {
+      zip(filesObj, (err, data) => {
+        resolve(data);
+      });
+    });
+
+    return await content;
```
Comment (JoostK) [inline adev/src/app/editor/download-manager.service.ts:68]: Stale `console.log` and any error is propagated; you'd at least need something like but I'm not sure what the error experience would be in that case

```suggestion
    return new Promise((resolve, reject) => {
      zip(filesObj, (err, data) => {
        if (err) {
          reject(err);
        } else {
          resolve(data);
        }
      });
    });
```

**Review by JoostK — COMMENTED** [review]
Drive-by comment. I think it makes more sense to change the `generateZip` implementation in `@angular/docs`, but can see this as an intermediate step to have a quick fix for adev.

File: adev/src/app/editor/download-manager.service.ts (hunk 2 above)
Comment (JeanMeche) [reply adev/src/app/editor/download-manager.service.ts:68]: The medium-term goal is the bring back the docs package into the framework repo. 
So I went with this instead. 

Also we didn't have any error handling ATM, so I went without it.

**Review by devversion — APPROVED** [review]
LGTM, but would be good to move this into `@angular/docs` + the commit message should mention why we switch from A to B

**Comment by JeanMeche** [comment]
I'll actually supersed it with angular/dev-infra#2165

**Pull request closed without merging**